### DISCUSSION TITLE
question about data structure

### DISCUSSION THREAD

**Discussion by alexandris-cyber** [community/community#21512] [Other Feature Feedback, Questions, & Ideas]
<p>This image is not mine.It is from wikipedia.May someone explain me with a simple way what are keys, hash functions and buckets in data structure?</p>
<p>thank you</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/c/c465f7e2b2681d9a9e4491da389090b9b7e3ac21.jpeg" data-download-href="/uploads/short-url/s1q37jmsSEZUDnLJmeLRdeTCavT.jpg?dl=1" title="programming question 2.jpg" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/61056385/181065072-34d7087e-a11f-42c4-b5d1-63638970e51f.jpeg" alt="programming question 2.jpg" data-base62-sha1="s1q37jmsSEZUDnLJmeLRdeTCavT" width="684" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/c/c465f7e2b2681d9a9e4491da389090b9b7e3ac21_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">programming question 2.jpg</span><span class="informations">736×538 80.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by alexandris-cyber** [comment]
<p>I havent any backround in computer programming</p>

**Comment by fire-eggs** [comment, marked as answer]
<p>You might try <a href="https://medium.com/omarelgabrys-blog/hash-tables-2fec6870207f" rel="nofollow noopener">this</a>. But it might be a little “low level”.</p>

**Comment by fire-eggs** [comment]
<p>If that page is too complex, let me know and I’ll try to explain on my own.</p>

**Comment by alexandris-cyber** [comment]
<p>thank you very much fire-eggs!!!I understand that you help me with the best of your abillities</p>
